### ISSUE TITLE
`require` support

### ISSUE THREAD

**Issue by paulmelnikow** [lydell/eslint-plugin-simple-import-sort#6]
I have to say, I love how this project works! It nails the developer experience, and just works without config, all the way down to the error messages.

Unfortunately I'm coming from a mixed project, https://github.com/badges/shields (the front end uses ESM and the back end uses `require`) so I'm using `import/order`.

It's really clear from the readme that `require` isn't supported, though wanted to say I would definitely use it if it were added!

Thanks so much for a great project!

**Comment by lydell** [comment]
Thanks for the kind words!

I would love a really good `require` sorter too! However, I'm not interested in making one, because these days the majority of the code I write uses `import` only, and as such I don't want to spend the time on dealing with all the complexity of `require`.

If you make or find a good `eslint-plugin-simple-require-sort` plugin – feel free to share it and I'll happily link to it!

I'm also kind of hoping that `import` will start being a thing in Node.js code before a `eslint-plugin-simple-require-sort` plugin comes into existence, so that sorting `require` becomes less and less relevant :)

For now I'm sticking with `import/order` which works OK-ish.

**Comment by paulmelnikow** [comment]
> If you make or find a good `eslint-plugin-simple-require-sort` plugin – feel free to share it and I'll happily link to it!

Kind of hoping someone comes along and does just this!

> I'm also kind of hoping that `import` will start being a thing in Node.js code before a `eslint-plugin-simple-require-sort` plugin comes into existence, so that sorting `require` becomes less and less relevant :)

Yea, I agree, we might get there relatively soon, at least in applications which only need to support a single Node version.

However in libraries there's a lot of diversity of opinion. Many libraries have a policy of targeting all the LTS versions of Node. For example, the latest release of Nock supports Node 6, and it's only in beta releases where Node 6 support has been dropped. Express takes a particularly (conservative? liberal?) approach: according to their `package.json` they still supports 0.10.0! That might be a little extreme, but keep in mind that it's not an obscure package: express is a big project with a big ecosystem of dependencies, all of which are basically following that rule.

TLDR, I think `require()` is going to be around for a while.

Of course I'm not trying to convince you to do work you don't want to do; just wanting to encourage someone who may be feeling this need more acutely, that this is worth taking on.

> For now I'm sticking with `import/order` which works OK-ish.

👍

**Comment by lydell** [comment]
> I think `require()` is going to be around for a while.

Yeah, me too. Forever, actually. But at some point (in a couple of years) hopefully all _new_ packages will be written with `import`.

**Comment by namnm** [comment]
Not asking for supporting `require` but just want to say thank you for the plugin. I was dead from using a tool with jscodeshift until now. Keep doing your great work!
